src/views/Stage.vue

```diff
@@ -231,60 +225,67 @@ export default class Map extends Vue {
   }
 
   private nextTurn(animation: string) {
-    this.animation = animation
-    // Vue.nextTick(() => {
-    this.loopTurn()
-    // })
-    // this.generateGrid()
-
-    this.placeEntities()
-    // this.calculateObserveRange()
+    this.animation[0] = animation
+    this.calculateObserveRange()
+    this.placeEntity(this.level.getPlayer())
+    Vue.nextTick(() => {
+      this.AnimateCamera(true)
+      setTimeout(() => {
+        this.loopTurn()
+      }, 500);
+    })
   }
 
   private loopTurn() {
+    let ent: Entity = this.aiHandler.peekNextTurn()
+    this.displaceEntity(ent)
     if (this.aiHandler.nextTurn()) {
-     // Vue.nextTick(() => {
-        this.loopTurn()
-      // })
+      Vue.nextTick(() => {
+        setTimeout(() => {
+          this.loopTurn()
+        }, 500);
+      })
     }
+    this.placeEntity(ent)
   }
 
   private placeEntities() {
-    this.gridRenderArray.forEach((gridBlock) => {
+    for (const gridBlock of this.gridRenderArray) {
```

Comment:
is die benodig? Word hulle nie een vir een displace nie of is dit nog todo?

Reply:
placeEntities is 'n function wat all die entities displace, en hulle weer place, hy word net in die begin geroep, nie na else action nie